tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1437 @@
+                      parameterFile: Option[String] = None,
+                      blocking: Boolean = false,
+                      result: Boolean = false,
+                      expectedExitCode: Int = Accepted.intValue)(implicit wp: WskProps): RestResult = {
+    super.invokeAction(name, parameters, parameterFile, blocking, result, expectedExitCode = expectedExitCode)
+  }
+}
+
+class WskRestTrigger
+    extends RunWskRestCmd
+    with ListOrGetFromCollectionRest
+    with DeleteFromCollectionRest
+    with HasActivationRest
+    with BaseTrigger {
+
+  override protected val noun = "triggers"
+
+  /**
+   * Creates trigger. Parameters mirror those available in the REST.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def create(name: String,
+                      parameters: Map[String, JsValue] = Map(),
+                      annotations: Map[String, JsValue] = Map(),
+                      parameterFile: Option[String] = None,
+                      annotationFile: Option[String] = None,
+                      feed: Option[String] = None,
+                      shared: Option[Boolean] = None,
+                      update: Boolean = false,
+                      expectedExitCode: Int = OK.intValue)(implicit wp: WskProps): RestResult = {
+
+    val (ns, triggerName) = this.getNamespaceActionName(name)
+    val path = Path(s"$basePath/namespaces/$ns/$noun/$triggerName")
+    val (params, annos) = getParamsAnnos(parameters, annotations, parameterFile, annotationFile, feed)
+    var bodyContent = JsObject("name" -> name.toJson, "namespace" -> s"$ns".toJson)
+
+    if (!update) {
+      val published = shared map { s =>
+        s
+      } getOrElse false
+      bodyContent = JsObject(
+        bodyContent.fields + ("publish" -> published.toJson,
+        "parameters" -> params, "annotations" -> annos))
+    } else {
+      bodyContent = shared map { s =>
+        JsObject(bodyContent.fields + ("publish" -> s.toJson))
+      } getOrElse bodyContent
+
+      val inputParams = convertMapIntoKeyValue(parameters)
+      if (inputParams.elements.size > 0) {
+        bodyContent = JsObject(bodyContent.fields + ("parameters" -> params))
+      }
+      val inputAnnos = convertMapIntoKeyValue(annotations)
+      if (inputAnnos.elements.size > 0) {
+        bodyContent = JsObject(bodyContent.fields + ("annotations" -> annos))
+      }
+    }
+
+    val resp =
+      if (update) requestEntity(PUT, path, Map("overwrite" -> "true"), bodyContent.toString)
+      else requestEntity(PUT, path, body = bodyContent.toString)
+    val result = new RestResult(resp.status, getRespData(resp))
+
+    val feedInput = feed map { f =>
+      f
+    } getOrElse null
+
+    if ((feed == null) || (result.statusCode != OK)) {
```

Comment:
How would `feed` be null here if it is a `Some` or `None`?

Reply:
It should be feedInput, but I will change part anyway.